### PR TITLE
add debounce for GSelect and RemoteSelect

### PR DESCRIPTION
**Pull request by maskin25**
# What this PR does

Fix performance Issue in GSelect component when searching

## Which issue(s) this PR fixes

https://github.com/grafana/oncall/issues/1628

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

**Review by joeyorlando — APPROVED** [review]
niceee 🤘

File: grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx [hunk 1]
```diff
@@ -127,7 +124,7 @@ const RemoteSelect = inject('store')(
         isSearchable={showSearch}
         value={value}
         onChange={onChangeCallback}
-        defaultOptions={options}
+        defaultOptions
```
Comment (joeyorlando) [inline grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx:127]: what is this change for?

File: grafana-plugin/src/models/grafana_team/grafana_team.ts [hunk 2]
```diff
@@ -29,7 +30,9 @@ export class GrafanaTeamStore extends BaseStore {
 
   @action
   async updateItems(query = '') {
-    const result = await this.getAll();
+    const result = await makeRequest(`${this.path}`, {
+      params: { search: query },
+    });
```
Comment (joeyorlando) [inline grafana-plugin/src/models/grafana_team/grafana_team.ts:35]: why include this change in this PR?

File: grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx (hunk 1 above)
Comment (maskin25) [reply grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx:127]: from [specs](https://developers.grafana.com/ui/latest/index.html?path=/docs/forms-select--auto-menu-placement)

```
defaultOptions | When specified as boolean the loadOptions will execute when component is mounted
```
it exactly what we need

File: grafana-plugin/src/models/grafana_team/grafana_team.ts (hunk 2 above)
Comment (maskin25) [reply grafana-plugin/src/models/grafana_team/grafana_team.ts:35]: search in teams GSelect doesn't work, it's related to current issue, so decided to fix this too

File: grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx (hunk 1 above)
Comment (joeyorlando) [reply grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx:127]: gotcha, thanks for clarifying 👍

**Pull request merged**